pkg/planner/core/plan_cache.go

```diff
@@ -447,6 +448,12 @@ func rebuildRange(p Plan) error {
 			return err
 		}
 	case *PointGetPlan:
+		if x.TblInfo.GetPartitionInfo() != nil {
+			if fixcontrol.GetBoolWithDefault(sctx.GetSessionVars().OptimizerFixControl, fixcontrol.Fix33031, false) {
+				sctx.GetSessionVars().StmtCtx.UseCache = false
```

Comment:
Please use `stmtctx.SetSkipPlanCache` as well.

Reply:
I changed the calling function to use `SetSkipPlanCache` instead, so now only returning an error. If I would use `SetSkipPlanCache` here, it would duplicate the warnings otherwise.

Reply:
A bit too early, I will keep the `UseCache = false` here, and not changing the calling function to use `SetSkipPlanCache` instead of `AppendWarning`.

Reply:
We agreed on just remove the UseCache = false and accept possible duplicate warnings.